typescript/utils/src/__tests__/sleep.test.ts

```diff
@@ -0,0 +1,14 @@
+import { sleep } from "../sleep";
+
+jest.useFakeTimers();
+
+describe("sleep", () => {
+  it("waits for the specified amount of time", async () => {
+    const resolve = jest.fn();
+    const reject = jest.fn();
+    sleep(100).then(resolve).catch(reject);
+    jest.advanceTimersByTime(100);
```

Comment:
```suggestion
    jest.advanceTimersByTime(50);
    await Promise.resolve();
    expect(resolve).not.toHaveBeenCalled();
    jest.advanceTimersByTime(50);
```

This works as expected, and now tests both before and after the sleep delay.

Reply:
Nice ❤️ Thank you!